data/DataError.go

```diff
@@ -0,0 +1,88 @@
+package data
+
+import "fmt"
+
+//ErrorSet contains a list of Error's
+type ErrorSet struct {
+	errors []*Error
+}
+
+//NewErrorSet creates an initialised ErrorSet
+func NewErrorSet() *ErrorSet {
+	return &ErrorSet{}
+}
+
+//AppendError appends an error to the ErrorSet
+func (e *ErrorSet) AppendError(err *Error) {
+	if err == nil || err.IsEmpty() {
+		return
+	}
+	e.errors = append(e.errors, err)
+	return
+}
+
+//Error returns a string representation of the errors
+func (e *ErrorSet) Error() string {
+
+	errorsStr := ""
+	if e != nil {
+		for i := range e.errors {
+			if e.errors[i] != nil {
+				errorsStr = fmt.Sprintln(errorsStr, e.errors[i].Error())
+			}
+		}
+	}
+
+	return errorsStr
+}
+
+//Error contains a list or errors and asscociated data
+type Error struct {
+	errors []error
+	data   map[string]interface{}
```

Comment:
What do we need the data here for?  Is the array for the potential multiple validation errors?